### PR TITLE
Fix Jax WMT workload

### PR DESCRIPTION
**Pull request by runame**
I wanted to wait until the hanging issue (#105) is fixed, but since it might just require a more recent jax version, I think this fix here can already be merged. See #105 for an explanation.

Besides the fix, I have also refactored the reference submission to use `workload.loss_fn()`.

### CODE REVIEW

**Review by znado — APPROVED** [review]

**Comment by znado** [comment]
LGTM assuming we aren't tuning label smoothing

File: reference_submissions/wmt/wmt_jax/submission.py [hunk 1]
```diff
@@ -98,49 +96,32 @@ def init_optimizer_state(workload: spec.Workload,
 
 @functools.partial(
     jax.pmap,
-    in_axes=(None, None, 0, 0, 0, None, 0),
```
Comment (znado) [inline reference_submissions/wmt/wmt_jax/submission.py:101]: lets revert all of the changes to this file, because we will need them later when we tune label smoothing in the v2 of the target setting runs (if we ever want to remove this code in the future, we can always git cherry-pick this commit).

**Review by znado — CHANGES_REQUESTED** [review]

File: reference_submissions/wmt/wmt_jax/submission.py (hunk 1 above)
Comment (runame) [reply reference_submissions/wmt/wmt_jax/submission.py:101]: Maybe it makes sense to just add an additional `label_smoothing` argument to `workload.loss_fn()` if we want to tune label smoothing for all workloads anyway. Alternatively, we would have to allow one-hot labels as input to `workload.loss_fn()`; currently it says [Dense (not one-hot) labels](https://github.com/mlcommons/algorithmic-efficiency/blob/main/algorithmic_efficiency/spec.py#L225).

 In any case the submissions should use `workload.loss_fn()`, since it is part of the definition of a workload, i.e. if a user is not using it, their submission could be considered invalid.

Comment (znado) [reply reference_submissions/wmt/wmt_jax/submission.py:101]: sgtm. I can add label_smoothing and dropout to the workload API in a follow-up PR as we discussed

**Review by znado — APPROVED** [review]

File: reference_submissions/wmt/wmt_jax/submission.py (hunk 1 above)
Comment (runame) [reply reference_submissions/wmt/wmt_jax/submission.py:101]: I wanted to suggest the same, I can add them in the target-setting run PR, since that is where they are needed anyway.
